test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoTest.java

```diff
@@ -86,14 +90,55 @@ public void canCrudOneSchedulePlan() {
     
     @Test
     public void getAllSchedulePlans() {
-        TestABSchedulePlan abPlan = new TestABSchedulePlan();
-        TestSimpleSchedulePlan simplePlan = new TestSimpleSchedulePlan();
+        SchedulePlan abPlan = TestABSchedulePlan.create();
+        SchedulePlan simplePlan = TestSimpleSchedulePlan.create();
         
-        schedulePlanDao.createSchedulePlan(abPlan);
-        schedulePlanDao.createSchedulePlan(simplePlan);
+        SchedulePlan plan1 = schedulePlanDao.createSchedulePlan(abPlan);
+        SchedulePlan plan2 = schedulePlanDao.createSchedulePlan(simplePlan);
         
-        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(studyIdentifier);
-        assertEquals("2 plans exist", 2, plans.size());
+        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyIdentifier);
+        assertEquals(getSchedulePlanGuids(plan1, plan2), getSchedulePlanGuids(plans));
+    }
+    
+    @Test
+    public void filtersSchedulePlans() {
+        Set<String> guids = Sets.newHashSet();
+        Set<String> oneGuid = Sets.newHashSet();
+        
+        List<SchedulePlan> plans = TestUtils.getSchedulePlans();
+        SchedulePlan plan = schedulePlanDao.createSchedulePlan(plans.get(0));
+        guids.add(plan.getGuid());
+        
+        plan = schedulePlanDao.createSchedulePlan(plans.get(1));
+        guids.add(plan.getGuid());
+        oneGuid.add(plan.getGuid());
+
+        plan = schedulePlanDao.createSchedulePlan(plans.get(2));
+        guids.add(plan.getGuid());
+        
+        // No known client, all the guids are returned
+        plans = schedulePlanDao.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyIdentifier);
+        assertEquals(guids, getSchedulePlanGuids(plans));
```

Comment:
As is, this matches the schedule plan guids exactly. This would break if there's another test being run in parallel (such as with parallel builds). I think what you actually want is getSchedulePlanGuids(plans).containsAll(guids)